### DISCUSSION TITLE
Impossible to add two or more dagruns (for one dag) with the same execution_date.

### DISCUSSION THREAD

**Discussion by marcin-adamski** [apache/airflow#37133] [Q&A]
Hi,
I have a question. I would like to start (by using API) many dagruns based on one dag which differ by the conf parameters (provided by API call). I found in the "post_dag_run" function  (https://github.com/apache/airflow/blob/main/airflow/api_connexion/endpoints/dag_run_endpoint.py) at line: 333. that the execution_date of a new dagrun is tested if it is equal to some existing dagrun. 
So, I understand that it is impossible to add two or more dagruns with the same execution_date. 
Am I right?  Is it a feature?

A workaround could be to use many dags but the management will be not so easy in my case, the number of dag will change in time.
Can you suggest a better solution?

Best regards,
Marcin.

**Comment by potiuk** [comment]
Yes. It's something that remained as unique constraint from times where we did not have dag_run. There were discussions about removing this limit and likely it will happen eventually. 

One workaround might be to add milliseconds to the execution_date to make it unique - we had even some deprecated API that allowed that in the past with a special flag as far as I remember.
